### ISSUE TITLE
Expose Shaped Sweeping to Protoflux (BoxSweep, SphereSweep, CapsuleSweep)

### ISSUE THREAD

**Issue by Foohy** [Yellow-Dog-Man/Resonite-Issues#2366]
### Is your feature request related to a problem? Please describe.

I am currently working on a custom locomotion module, which is requiring that I do some in-depth tracing for player collisions. While single line traces work okay for checking points through space, I need to determine if the player's entire collision shape (eg. Capsule collision or box collision) will collide with anything if I move them in a particular direction.

### Describe the solution you'd like

Three new protoflux nodes, which function nearly identically to RaycastOne (returning a single RaycastHit after hitting something): 
`BoxSweepOne(origin, rotation, direction, size) -> RaycastHit`
`CapsuleSweepOne(origin, rotation, direction, radius, length) -> RaycastHit`
`SphereSweepOne(origin, direction, radius) -> RaycastHit`

I believe Bepu has existing support for sweeping shapes, and I suspect FrooxEngine may already have some of those hooked up already, just not exposed to the user in any way.

### Describe alternatives you've considered

It's never perfect, but you can get close-ish by just throwing a whole bunch of traces in the shape of the top and bottom of a box at it, but this obviously comes with a ton of cost overhead versus a single sweep trace.

### Additional Context

While I'm using this for locomotion, swept shape queries can come in handy for a heck of a lot more than that. Weapons that fire "thicker" bullets, more efficient area checks, really anything that involves querying the collision environment in a more broad way would benefit from these nodes being exposed.

### Requesters

Foohy

**Comment by BaxterOttoman** [comment]
I also would love this and support the idea, even though it will probably be something that is going to have to come with collections, by way of a sweep having many possible collisions and a ray cast only returning one.

**Comment by BlueCyro** [comment]
> I also would love this and support the idea, even though it will probably be something that is going to have to come with collections, by way of a sweep having many possible collisions and a ray cast only returning one.

I see no reason this can't simply return the first thing it hit much like RaycastOne already does. Initially it wouldn't be as robust, but it would still be useful for determining whether something exists within the path of an extruded volume rather than an infinitely thin line.

**Comment by Frooxius** [comment]
Yeah I agree, we can add SweepOne versions for these nodes, same way RaycastOne works - we already have those functions internally, so it's mostly just wrapping them.

The whole idea of RaycastOne is that it allows the function to be used without needing collections yet. Once we have collections, we can introduce RaycastAll/SweepAll versions that give you a collection of results.

**Comment by BlueCyro** [comment]
Since it's easy enough to do I'll try and finish this by tomorrow or something :3
